pavelib/quality.py

```diff
@@ -1,12 +1,89 @@
 """
 Check code quality using pep8, pylint, and diff_quality.
 """
+from __future__ import print_function
 from paver.easy import sh, task, cmdopts, needs, BuildFailure
 import os
 import re
 
 from .utils.envs import Env
 
+# These are all of the top-level directories to be checked.
+ALL_SYSTEMS = {
+    'lms',
+    'cms',
+    'common',
+    'openedx',
+    'pavelib',
+    'scripts',
+    'docs',
+}
+
+# These are all of the second-level directories to be checked.
+ALL_SUBSYSTEMS = {
+    'djangoapps',
+    'lib',
+}
+
+
+def _get_path_list(system):
+    """
+    Gather a list of subdirectories within the system
+
+    :param system: the system directory to search; e.g. 'lms', 'cms'
+    :returns: a list of system subdirectories to be linted
+    """
+    paths = [system]
+
+    for directory in ALL_SUBSYSTEMS:
+        try:
+            directories = os.listdir(os.path.join(system, directory))
+        except OSError:
+            pass
+        else:
+            paths.extend([
+                directory
+                for directory in directories
+                if os.path.isdir(os.path.join(system, directory, directory))
+            ])
+
+    path_list = ' '.join(paths)
+    return path_list
+
+
+def _get_python_path_prefix(system):
+    """
+    Build a string to specify the PYTHONPATH environment variable
+
+    :param system: the system directory to search; e.g. 'lms', 'cms'
+    :returns str: a PYTHONPATH environment string for the command line
+    """
+    paths = {
+        system,
+        os.path.join(system, 'lib'),
+        os.path.join(system, 'djangoapps'),
```

Comment:
Is there a reason to hardcode lib & djangoapps rather than using the ALL_SUBSYSTEMS variable? Also in the get_path_list function you make sure that these subdirs exist - is that a concern/convention we should follow here?